autometa/common/exceptions.py

```diff
@@ -39,41 +38,118 @@ def __init__(self, value):
+
     def __str__(self):
-        return f"{self.value}\n\n{issue_request}"
+        return f"{self.value}\n\n{self.issue_request}"
 
 
 class KmerFormatError(AutometaException):
-    """KmerFormatError exception class."""
+    """
+    KmerFormatError exception class.
+
+    Parameters
+    ----------
+    AutometaException : class
+        Base class for other exceptions
+    """
 
     def __init__(self, fpath):
-        super(AutometaException, self).__init__(fpath)
         self.fpath = fpath
 
     def __str__(self):
+        """
+        Operator overloading to return the path to k-mers frequency tab-delimited table or
+        k-mers normalized tab-delimited table
+
+        Returns
+        -------
+        str
+            Path to k-mers frequency tab-delimited table or k-mers normalized tab-delimited table
+        """
         return (
             f'{self.fpath} does not contain a "contig" column. '
             "Ensure the k-mer matrix was properly generated."
         )
 
 
 class KmerEmbeddingError(AutometaException):
-    """KmerEmbeddingError exception class."""
+    """
+    KmerEmbeddingError exception class.
+
+    Parameters
+    ----------
+    AutometaException : class
+        Base class for other exceptions
+    """
 
     def __init__(self, value):
         self.value = value
 
     def __str__(self):
+        """
+        Operator overloading to return the text message written while raising the error,
+        rather than the message of __str__ by base exception
+
+        Returns
+        -------
+        str
+            Message written alongside raising the exception
+        """
         return self.value
 
 
 class BinningError(AutometaException):
-    """BinningError exception class."""
+    """
+    BinningError exception class.
+
+    Parameters
+    ----------
+    AutometaException : class
+        Base class for other exceptions
+    """
+
+    def __init__(self, value):
+        self.value = value
+
+    def __str__(self):
+        """
+        Operator overloading to return the text message written while raising the error,
+        rather than the message of __str__ by base exception
+
+        Returns
+        -------
+        str
+            Message written alongside raising the exception
+        """
+        return self.value
+
+
+class DatabaseOutOfSyncError(AutometaException):
+    """
+    Raised when NCBI databases nodes.dmp, names.dmp and merged.dmp are out of sync with each other
+
+    Parameters
+    ----------
+    AutometaException : class
+        Base class for other exceptions
+    """
 
     def __init__(self, value):
         self.value = value
 
     def __str__(self):
+        """
+        Operator overloading to return the text message written while raising the error,
+        rather than the message of __str__ by base exception
+
+        Returns
+        -------
+        str
+            Message written alongside raising the exception
+        """
```

Comment:
I don't think the `__init__` or `__str__` are necessary, unless you want to wrap the message with another message. Perhaps the links to the database urls to download the databases? Or the autometa command to download/update the databases?

Reply:
Yeah, that's a nice idea. However, I can emit that. However, I don't know the autometa command to download the databases